### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by zemola**
**React capstone project**
For this capstone project, the below requirements were observed.

- An API was chosen to base the development of the webapp on. The API allows you to get numeric values based on a parameter.
- The project design was based on the template given.
-When the page loads, the webapp shows the list of categories that could be filtered out by the category name.

- When the user clicks (or taps) on a category item, the application navigates to the details page.

- In the details page, the webapp retrieves data from the API to show detailed data in the category.

- When the user clicks on the "Back" button (<), the user navigates to the home page.
- Create unit tests for pure functions in Redux code.

- Create integration tests for your application using the React Testing Library.

- Mock the access to the API, so that tests don't send actual requests.

- Mock the connection to the Redux Store.

- The project is a single-page application (SPA) built with React and Redux.

- The data retrieved from the API is stored in the Redux store.

- Filter the data retrieve from the API using a Filter stateless component.

- A video was recorded presenting the features of the project built. It is a video with a length of 5 minutes.

### CODE REVIEW

File: src/components/DetailList.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  useEffect(() => {
+    if (dataCovid.country.length === 0) {
+      dispatch(getCovidFromApi());
+    }
+  }, []);
+  const todayDate = new Date().toISOString().slice(0, 10);
+  const handleCountry = () => {
+    const country = useParams();
+    const hello = country.country;
+    // const num = document.getElementById('date').value;
```
Comment (bimbolabuari) [inline src/components/DetailList.js]: Let's remove all commented code, to make the code cleaner and professional 🙂 

Kindly check for other occurrences 👍

File: src/components/Details.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            </div>
+            <div className="sor">
+              <p>Yesterday Open Cases:</p>
+              <b className="dat-type">{cou.data.yesterday_open_cases}</b>
+            </div>
+            <div className="sor">
+              <p>Yesterday Recovered:</p>
+              <b className="dat-type">{cou.data.yesterday_recovered}</b>
+            </div>
+          </div>
```
Comment (bimbolabuari) [inline src/components/Details.js:92]: Currently, I am unable to check each countries details with your deployed webpage. It is required that the project is deployed properly, the home and detail page works as expected.

image

Kindly fix this 👍

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// import Details from './components/Details';
+import './App.css';
+
+const App = () => (
+  <BrowserRouter>
+    <Nav />
+    <Routes>
+      <Route exact path="/" element={<Home />} />
+      <Route path="/:country" element={<DetailLists />} />
+      {/* <Route path="/details" element={<Details />} /> */}
```
Comment (bimbolabuari) [inline src/App.js]: Let's remove all commented code, to make the code cleaner and professional 🙂

Kindly check for other occurrences 👍

File: README.md [hunk 4]
```diff
@@ -1 +1,69 @@
...
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/zemola/metrics-webapp/issues)).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+Thanks to my teammates, learning and coding partners.
```
Comment (bimbolabuari) [inline README.md]: As required, Kindly give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @zemola, 

You have done a good job so far 🚀  

## Highlights 🎉 

- Nice Video Presentation
- Deploys the application using Netlify
- The project contains a `Home` page and `Detail` page.
- The project is a single-page application (SPA) using `react-router-dom` to serve each page without reloading.

There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

**Comment by zemola** [comment]
Dear Reviewer,
Kindly note I had to change my live link reason been that my account with netlify got suspended, attached is the image evidence.
image

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave: @zemola  , 

appr

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- Your linters are passing :heavy_check_mark:  
- Your README file looks very professional :+1: 
- You did a great job :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**